Libraries/src/Amazon.Lambda.Annotations/README.md

```diff
@@ -68,37 +187,141 @@ public class Functions
+      "Metadata": {
+        "Tool": "Amazon.Lambda.Annotations",
+        "SyncedEvents": [
+          "RootGet"
+        ]
+      },
+      "Properties": {
+        "Runtime": "dotnet6",
+
+...
+        "Role": {
+          "Fn::GetAtt": [
+            "LambdaRoleParameter",
+            "Arn"
+          ]
+        }
+      }
+    },
+```
+
+## Amazon API Gateway example
+
+This example creates a REST API through Amazon API Gateway that exposes the common arithmetic operations. 
+
+To avoid putting business logic inside the REST API a separate calculator service is created to encapsulated the logic of the arithmetic operations. Here is both the 
+calculator service's interface and default implementation.
+
+```csharp
+public interface ICalculatorService
+{
+    int Add(int x, int y);
+
+    int Substract(int x, int y);
+
+    int Multiply(int x, int y);
+
+    int Divide(int x, int y);
+}
+
+public class DefaultCalculatorService : ICalculatorService
+{
+    public int Add(int x, int y) => x + y;
+
+    public int Substract(int x, int y) => x - y;
+
+    public int Multiply(int x, int y) => x * y;
+
+    public int Divide(int x, int y) => x / y;
+}
+```
+
+The startup class contains the `LambdaStartup` attribute identifying it as the class to configure the services registered in the dependency injection framework. 
+Here the `ICalculatorService` is registered as a singleton service in the collection of services.
 
+```csharp
+[LambdaStartup]
+public class Startup
+{
+
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddSingleton<ICalculatorService, DefaultCalculatorService>();
+    }
+}
 ```
 
-  "Parameters": {
-    "LambdaExecutionRole": {
-      "Type": "String"
+Since the `ICalculatorService` is registered as a singleton the service is injected into the Lambda function via the constructor. 
+If the registered service is registered as scoped or transient and a new instance is need for each Lambda invocation then the
+`FromServices` attribute should be used on a method parameter of the Lambda function. 
+
+```csharp
+public class Functions
+{
+    ICalculatorService _calculatorService;
+    public Functions(ICalculatorService calculatorService)
+    {
+        _calculatorService = calculatorService;
     }
-  },
 
+    ...
 ```
 
-The above two examples when used together will the use the value of LambdaRoleParameter as the role during deployment.
+For each arithmetic operation a separate C# method is added containing the `LambdaFunction` attribute. The `LambdaFunction` attribute
+ensures the dependency injection framework is hooked up to the Lambda function and the Lambda function will be declared in the 
+CloudFormation template. 
 
-## Source Generator
+Since these Lambda functions are responding to API Gateway events the `HttpApi` attribute is added 
+to registered the event source in CloudFormation along with the HTTP verb and resource path. The `HttpApi` attribute also enables 
```

Comment:
"to registered the event source in CloudFormation" should be "to register the event source..." ?

Reply:
Fixed